libcaf_core/caf/actor_registry.hpp

```diff
@@ -74,9 +87,23 @@ class actor_registry {
   /// Returns the actor associated with `key` or `invalid_actor`.
   strong_actor_ptr get(atom_value key) const;
 
+  /// Returns the actor associated with `key` or `invalid_actor`.
+  template<typename T>
+  T get(atom_value key) const {
+      return actor_cast<T>(get(key));
+  }
+
   /// Associates given actor to `key`.
   void put(atom_value key, strong_actor_ptr value);
 
+  /// Associates given actor to `key`.
+  template<typename T>
+  void put(atom_value key, T& value) {
+      // using reference here and before to allow putting a scoped_actor without calling .ptr()
+      // also prevents calling this version from non-templated put as (key, nullptr)
+      put(key, actor_cast<strong_actor_ptr>(value));
```

Comment:
I am not sure if we should allow putting `scoped_actors` in there. Maybe @Neverlord can give some advise.

Reply:
I think it wouldn't hurt as when the actor is deconstructed it's removed from the registry anyway. But communicating back to an "initial" actor, say, created in main() is a nice use case.